### ISSUE TITLE
Whip

### ISSUE THREAD

**Issue by bibbybabby** [studio-minus/ppg-bugs-and-requests#1479]
**Feature description**
Rope on a stick that is used to whip humans

**Feature purpose**
Custom made whips clip through and get stuck inside because unity fuckery, also no whip crack sound

**Additional context**
If a rope isnt possible maybe add a riding crop?

**Comment by DindinYT37** [comment]
A Rope with collision would be really janky and buggy, so I doubt this would be possible. The Reason a Rope wire is possible is because it has no collision (at least I think so).

**Comment by ppgplayer** [comment]
> This would be really janky and buggy, so I doubt this would be possible. The Reason a Rope wire is possible is because it has no collision (at least I think so).

Man stfu I just wanna torture virtual characters in NEW and INNOVATIVE ways

**Comment by DindinYT37** [comment]
> > This would be really janky and buggy, so I doubt this would be possible. The Reason a Rope wire is possible is because it has no collision (at least I think so).
> 
> Man stfu I just wanna torture virtual characters in NEW and INNOVATIVE ways

No, you can stfu. You're just being ignorant of my simple explanation about why it would probably be extremely janky and buggy, which is why it likely won't happen.

**Comment by ppgplayer** [comment]
> > > This would be really janky and buggy, so I doubt this would be possible. The Reason a Rope wire is possible is because it has no collision (at least I think so).
> > 
> > 
> > Man stfu I just wanna torture virtual characters in NEW and INNOVATIVE ways
> 
> No, you can stfu. You're just being ignorant of my simple explanation about why it would probably be extremely janky and buggy, which is why it likely won't happen.

![image](https://user-images.githubusercontent.com/98086053/173265522-de010fc8-19bf-467f-a5df-3d44388c21b2.png)

**Comment by ppgplayer** [comment]
imma be honest guys idk how ME and this guy got that low in life where we argue on GITHUB

Still think he a bitch tho

**Comment by DindinYT37** [comment]
> imma be honest guys idk how ME and this guy got that low in life where we argue on GITHUB
> 
> Still think he a bitch tho

My man you just listed yourself first and called me an incel.

**Comment by ppgplayer** [comment]
> > imma be honest guys idk how ME and this guy got that low in life where we argue on GITHUB
> > Still think he a bitch tho
> 
> My man you just listed yourself first and called me an incel. Also, I don't know how we argue on GitHub, it's just that I'm commenting stuff and you keep replying insults or are telling me to shut up again.

🤓

**Comment by Drogo-man** [comment]
> [<img alt="@ppgplayer" width="28" height="28" src="https://avatars.githubusercontent.com/u/98086053?s=60&amp;v=4">](/ppgplayer)
> #### ** [ppgplayer](/ppgplayer) ** [10 minutes ago](#issuecomment-1153745933)

🤓

**Comment by JudicialShrimp** [comment]
> > > > This would be really janky and buggy, so I doubt this would be possible. The Reason a Rope wire is possible is because it has no collision (at least I think so).
> > > 
> > > 
> > > Man stfu I just wanna torture virtual characters in NEW and INNOVATIVE ways
> > 
> > 
> > No, you can stfu. You're just being ignorant of my simple explanation about why it would probably be extremely janky and buggy, which is why it likely won't happen.
> 
> ![image](https://user-images.githubusercontent.com/98086053/173265522-de010fc8-19bf-467f-a5df-3d44388c21b2.png)

Good sir or madam, I regret to inform you that your bingo board is lacking a free space, therefore cannot qualify as an officially recognized board.

I believe the whip would be relatively simple to implement with similar mechanics to a human shoulder/arm connection. Whip doesn't have collision with itself (except the base), but still has defined anchor points that stop the segments from simply falling through each other.

If it could work with a human arm, I don't see why it couldn't work roughly the same with a few extra segments? All we need here are blocks that rotate on a fixed axis and don't collide with each other to prevent jankiness, but they should still be able to effectively follow each other similar to how an arm can follow the shoulder, which should be enough to make a basic whip. You could have an extra option for the tip to auto-connect to things it touches as well, giving it more fluidity when playing around. Or steal the same system and make some kind of conveyor belt, idk.

**Comment by DindinYT37** [comment]
> Good sir or madam, I regret to inform you that your bingo board is lacking a free space, therefore cannot qualify as an officially recognized board.
> 
> I believe the whip would be relatively simple to implement with similar mechanics to a human shoulder/arm connection. Whip doesn't have collision with itself (except the base), but still has defined anchor points that stop the segments from simply falling through each other.
> 
> If it could work with a human arm, I don't see why it couldn't work roughly the same with a few extra segments? All we need here are blocks that rotate on a fixed axis and don't collide with each other to prevent jankiness, but they should still be able to effectively follow each other similar to how an arm can follow the shoulder, which should be enough to make a basic whip. You could have an extra option for the tip to auto-connect to things it touches as well, giving it more fluidity when playing around. Or steal the same system and make some kind of conveyor belt, idk.

That is not my bingo board.

What you're talking about is not a rope then, which is what he said. You "just" want a line of segments **simulating** a wire/rope.

Because what were talking about is an actual rope, not a few segments. Using a rope would be extremely janky but have less limitations than what you described. Even though what you meant would probably still be janky it'd be way better than what an actual rope.

*Also, zooi has said multiple times for us to not estimate how easy something is to implement. What I said is just that it's janky and buggy, what you said is that it would be easy to add - which we aren't supposed to do.*

**Comment by DindinYT37** [comment]
> > _Also, zooi has said multiple times for us to not estimate how easy something is to implement. What I said is just that it's janky and buggy, what you said is that it would be easy to add - which we aren't supposed to do._
> 
> I don't see any harm at all in estimating something, I'm merely throwing ideas out. The quickest and most hassle-free way would be to do something that allows you to recycle code/assets, which is what I was suggesting with the segment idea. In that regard, a decently believable whip could be created without having to implement an entirely new system for it.

You don't have to tell this to me, I don't decide it anyway. I was just informing you on what zooi has said before. Also, if you marked some text and press "Quote reply", it only quotes the marked text.

**Comment by JudicialShrimp** [comment]
> > > _Also, zooi has said multiple times for us to not estimate how easy something is to implement. What I said is just that it's janky and buggy, what you said is that it would be easy to add - which we aren't supposed to do._
> > 
> > 
> > I don't see any harm at all in estimating something, I'm merely throwing ideas out. The quickest and most hassle-free way would be to do something that allows you to recycle code/assets, which is what I was suggesting with the segment idea. In that regard, a decently believable whip could be created without having to implement an entirely new system for it.
> 
> You don't have to tell this to me, I don't decide it anyway. I was just informing you on what zooi has said before. Also, if you marked some text and press "Quote reply", it only quotes the marked text.

Oh, thanks for the quote tip! I'm rather new when it comes to using Github, so wasn't quite sure what it would do when I removed the image.
